src/java/com/google/cloud/bigquery/dwhassessment/hooks/test_utils/TestUtils.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2022-2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery.dwhassessment.hooks.test_utils;
+
+import static com.google.cloud.bigquery.dwhassessment.hooks.logger.LoggingHookConstants.QUERY_EVENT_SCHEMA;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.avro.file.DataFileStream;
+import org.apache.avro.generic.GenericData.Record;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.avro.io.DatumReader;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.ql.QueryPlan;
+import org.apache.hadoop.hive.ql.QueryState;
+import org.apache.hadoop.hive.ql.hooks.HookContext;
+import org.apache.hadoop.hive.ql.log.PerfLogger;
+import org.apache.hadoop.hive.ql.metadata.Hive;
+import org.apache.hadoop.hive.ql.parse.BaseSemanticAnalyzer;
+import org.apache.hadoop.hive.ql.parse.DDLSemanticAnalyzer;
+import org.apache.hadoop.hive.ql.plan.HiveOperation;
+
+/** Common utils for testing */
+public final class TestUtils {
+
+  public static final long QUERY_END_TIME = 9999L;
+  public static final String DEFAULT_QUERY_TEXT = "SELECT * FROM employees";
+  public static final String DEFAULT_QUERY_ID = "hive_query_id_999";
+
+  private final HiveConf conf;
+  private final QueryState queryState;
+  private final QueryPlan queryPlan;
+  private final HookContext hookContext;
+
+  public TestUtils(Hive hiveMock) throws Exception {
```

Comment:
Hiding the whole test setup in TestUtils constructor prevents us from extending or changing the part of this set up if needed. I would suggest removing the constructor and using a set of static methods `createDefault...` instead with a some redundancy in the test set up methods.

```java
@Before
public void setup() throws IOException {
  conf = new HiveConf();
  queryState = utils.createDefaultQueryState(conf);
  QueryPlan queryPlan = utils.createDefaultQueryPlan(queryState, hiveMock);
  hookContext = utils.createDefaultHookContext(queryState, queryPlan);
}
```

This will allow us to easily swap a part of the system and use other semantic analyzer or a query plan with other operation one.